Update master->main for versions repo refs

mmitche opened the pull request.

mmitche committed:
Update master->main for versions repo refs

MattGal commented on src/Microsoft.DotNet.VersionTools/Automation/GitHubVersionsRepoUpdater.cs:
```diff
@@ -137,11 +137,11 @@ public GitHubVersionsRepoUpdater(GitHubAuth gitHubAuth, GitHubProject project)
 
-                        GitTree tree = await client.PostTreeAsync(_project, masterSha, objects.ToArray());
-                        GitCommit commit = await client.PostCommitAsync(_project, message, tree.Sha, new[] { masterSha });
+                        GitTree tree = await client.PostTreeAsync(_project, mainRef, objects.ToArray());
```
should these be mainSha?

mmitche replied:
Wow good catch.

mmitche committed:
Fixup

mmitche commented:
/azp run

mmitche committed:
Pull dependency verification from master
The build process for buildtools will look paths in the versions repo that reference the old branch names. I do not think this is safe to chagne.
